platform/ios/MGLMapboxEvents.m

```diff
@@ -136,7 +195,26 @@ - (instancetype) init {
             
             [[NSUserDefaults standardUserDefaults] registerDefaults:defaultsToRegister];
         }
+    }
+}
+
++ (BOOL)isEnabled {
+    return ([[NSUserDefaults standardUserDefaults] boolForKey:@"MGLMapboxMetricsEnabled"] &&
+            [[NSUserDefaults standardUserDefaults] integerForKey:@"MGLMapboxAccountType"] == 0);
```

Comment:
`-[NSUserDefaults registerDefaults:]` guarantees that the defaults will always fall back to the passed-in dictionary values if unset. `-registerDefaults:` is called inside `+initialize`, which is guaranteed to be called when MapboxGL.a is loaded. I can certainly put in an assert for `MGLMapboxMetricsEnabled`, but it would add overhead.